src/Native/CpuMath/IntrinsicsUtils.cs

```diff
@@ -0,0 +1,796 @@
+﻿using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+using System;
+
+namespace Intrinsics
+{    public class IntrinsicsUtils
+    {
+        private static unsafe Vector128<float> Load1(float* src, int* idx)
+        {
+            return Sse.SetScalarVector128(src[idx[0]]);
+        }
+
+        private static unsafe Vector128<float> Load4(float* src, int* idx)
+        {
+            return Sse.SetVector128(src[idx[3]], src[idx[2]], src[idx[1]], src[idx[0]]);
+        }
+
+        private static Vector128<float> Rotate(Vector128<float> x)
+        {
+            return Sse.Shuffle(x, x, 0x39);
+        }
+
+        private static Vector128<float> RotateReverse(Vector128<float> x)
+        {
+            return Sse.Shuffle(x, x, 0x93);
+        }
+
+        // Warning: this operation changes the value of x => do not reuse x
+        private static unsafe void Store4(Vector128<float> x, float* dst, int* idx)
+        {
+            Sse.StoreScalar(dst + idx[0], x);
+            for (int i = 1; i <= 3; i++)
+            {
+                x = Rotate(x);
+                Sse.StoreScalar(dst + idx[i], x);
+            }
+        }
+
+        //// Multiply matrix times vector into vector.
+        //private static float MatMulASse(bool add, float[] matrix, float[] src, float[] dst)
```

Comment:
If these aren't complete, ready for use yet, we shouldn't include them as commented out as part of this PR.

Reply:
Right, normally please do'nt checkin commented code.